nginx-controller/nginx/configurator.go

```diff
@@ -766,6 +781,43 @@ func (cnf *Configurator) updatePlusEndpoints(ingEx *IngressEx) error {
 	return nil
 }
 
+func filterMasterAnnotations(annotations map[string]string) []string {
+	var removedAnnotations []string
+
+	for _, blacklistAnn := range masterBlacklist {
+		if _, ok := annotations[blacklistAnn]; ok {
+			removedAnnotations = append(removedAnnotations, blacklistAnn)
+			delete(annotations, blacklistAnn)
+		}
+	}
+
+	return removedAnnotations
+}
+
+func filterMinionAnnotations(annotations map[string]string) []string {
+	var removedAnnotations []string
+
+	for _, blacklistAnn := range minionBlacklist {
+		if _, ok := annotations[blacklistAnn]; ok {
+			removedAnnotations = append(removedAnnotations, blacklistAnn)
+			delete(annotations, blacklistAnn)
+		}
+	}
+
+	return removedAnnotations
+}
+
+func mergeMasterAnnotationsIntoMinion(minionAnnotations map[string]string, masterAnnotations map[string]string) {
+	var minionOverwrites []string
+
+	for key, val := range masterAnnotations {
+		if _, ok := minionAnnotations[key]; !ok {
```

Comment:
we also need to check that an annotation is not in the minion black list as well. If we don't do that, then an non-supported annotation will be merged into a minion. That will lead to the filterMinionAnnotations function returning that annotation in its removedAnnotations list. That will lead to an error message in the log, which should not happen!

Reply:
Good catch as well! Thanks.